.github/workflows/flags.yml

```diff
@@ -76,10 +76,10 @@ jobs:
       - name: Configue non-default flags for all components
         run: |
           cabal configure \
-              --constraint "hls-graph +embed-files +stm-stats" \
+              --constraint "haskell-language-server +pedantic" \
+              --constraint "hls-graph +embed-files +pedantic +stm-stats" \
               --constraint "ghcide +ekg +executable +test-exe" \
-              --constraint "hls-plugin-api -use-fingertree" \
-              --constraint "all +pedantic"
```

Comment:
This `all +pedantic` did not have any effect - it needs to be enabled on per-cabal-file basis.

Reply:
I didn't open any ticket about this in cabal, nor did I research why it doesn't work/whether it should work at all.
I just mistakenly assumed this works when I refactored this job.
Let me ask about it in cabal issue tracker later today.

Reply:
It turns out that the way to enable the flag for all local packages is via
`cabal configure --flag pedantic`

I'll make use of that in upcoming PR after I fix warnings in ghcide